Engine/Entities/Ants/Ant.cs

```diff
@@ -154,6 +154,16 @@ public void EmitFoodPheromone()
             Transform foodTransform = new(Transform.Position, 0, Vector2.One);
             World.AddEntity(new FoodPheromone(Name, foodTransform, World, PheromoneTimeSpan));
         }
+
+        private void CheckEntitiesInRegion<T>(int x, int y, IList list)
+        {
```

Comment:
This should not be in Ant but in World

Reply:
This method is an extracted method that allows for us to get the Entities in place x, y of type T that also are in distance perceptionrange of ant

Reply:
It would be slower to first iterate to get entities of type T and THEN only keep the ones who are in distance perceptionrange

Reply:
So it allows us to just perform the simple operation in one go, do you think the optimisation difference in iterating 2 times is negligable?